File: algorithms/1BitAnd2BitCharacters/solution.py
class Solution(object):
    def isOneBitCharacter(self, bits):
        """
        :type bits: List[int]
        :rtype: bool
        """
        index = 0
        while index < len(bits):
            if bits[index] == 1:
                if index + 2 == len(bits):
                    return False
                else:
                    index += 2
            else:
                index += 1
        
        return True

File: algorithms/1BitAnd2BitCharacters/Solution.java
class Solution {
    public boolean isOneBitCharacter(int[] bits) {
        int index = 0;
        
        while (index < bits.length) {
            if (bits[index] == 1) {
                if (index + 2 == bits.length) {
                    return false;
                } else {
                    index += 2;
                }
            } else {
                index++;
            }
        }
        
        return true;
    }
}